core/auth/anonymous-session.ts

```diff
@@ -0,0 +1,68 @@
+import { cookies } from 'next/headers';
+import { AnonymousUser } from 'next-auth';
+import { decode, encode } from 'next-auth/jwt';
+
+const anonymousCookieName = 'authjs.anonymous-session-token';
+
+export const anonymousSignIn = async (user: Partial<AnonymousUser> = { cartId: null }) => {
+  const secret = process.env.AUTH_SECRET ?? process.env.NEXTAUTH_SECRET;
+
+  if (!secret) {
+    throw new Error('AUTH_SECRET is not set');
+  }
+
+  const cookieJar = await cookies();
+  const jwt = await encode({
+    salt: anonymousCookieName,
+    secret,
+    token: {
+      user,
+    },
+  });
+
+  cookieJar.set(anonymousCookieName, jwt, {
+    secure: true,
+    sameSite: 'lax',
+    maxAge: 60 * 15, // 15 minutes
```

Comment:
Just curious, why is maxAge only 15 minutes? Isn't the default NextAuth maxAge ~30 days or so?